src/utils/logger.hpp

```diff
@@ -2,23 +2,58 @@
 
 #include <fmt/printf.h>
 
-extern int (*nrnpy_pr_stdoe_callback)(int, char*);
+using LoggerCallback = int(int, char*);
 
+class Logger {
+  public:
+    void setCallback(LoggerCallback* cb) {
+        callback = cb;
+    }
+
+    LoggerCallback* getCallback() const {
+        return callback;
+    }
+
+    template <typename... Args>
+    void info(const char* fmt, Args... args) {
+        if (callback) {
```

Comment:
If you want, you can finish the transformation by adding a default callback, i.e. something that always uses `std::cout`/`std::cerr` or `fprint`/`fmt::print` etc. That way you also separate the formatting from the outputting, which will automatically make it 100% consistent.

Reply:
Good idea, thanks.
Done.